src/workerd/api/kv.c++

```diff
@@ -266,16 +281,19 @@ jsg::Promise<jsg::Value> KvNamespace::list(jsg::Lock& js, jsg::Optional<ListOpti
 
       checkForErrorStatus("GET", response);
 
+      auto cacheStatus = response.headers->get(context.getHeaderIds().cfCacheStatus)
```

Comment:
@jasnell Is there a way to elide this extra copy? I tried creating a jsg::v8Str & passing that through, but then it failed claiming I was somehow creating a circular reference within the JSON? Not sure.

Reply:
Which copy -- the `kj::str(cs)`? If so, I'm not seeing a way to elide it given that we need `cacheStatus` to be usable in the `awaitIo` async callback below, after the response (and its headers) may have already been destructed.

It looks like we could plausibly avoid copying the `cacheStatus` in the case of a GET 404, but it's up to you whether that's worth worrying about since it makes the non-404 case *slightly* more complicated.

Reply:
I was thinking about creating the JS string up here directly rather than first copying it to a kj String & then copying it to JS. But I couldn't figure out how to get it to work.

Reply:
The other thing is, all these strings are constant and frequently should be just "HIT". So is there some way of efficiently interring the result (my guess is no but figured I'd ask).

Reply:
Figured out how to do a single copy into the v8 heap of cacheStatus for list for the cacheStatus.

@jasnell, Still wondering if I should be using `jsg::v8StrIntern` here for the cacheStatus (& in which case I think I'd need to make `cacheStatus` a `kj::Maybe<jsg::Value>` instead of `kj::Maybe<kj::String>` in `GetMetadataWithResult`).

Reply:
Discussed offline. `jsg::v8StrIntern` is the appropriate choice here because not using extended latin1.